mbuild/compound.py

```diff
@@ -69,53 +69,63 @@ def load(filename, relative_to_module=None, compound=None, coords_only=False,
     compound : mb.Compound
 
     """
+
+    if compound is None:
+        compound = Compound()
+
+    # First check if we are loading from an existing parmed or trajectory structure
+    type_dict = {pmd.Structure:compound.from_parmed,md.Trajectory:compound.from_trajectory}
+    if isinstance(filename_or_object, Compound):
+        return filename_or_object
+    for type in type_dict:
+        if isinstance(filename_or_object, type):
+            type_dict[type](filename_or_object,coords_only=coords_only, **kwargs)
+            return compound
```

Comment:
This is out of the scope of this PR but this loop provides an opportunity to sanitize inputs that I don't think we are currently doing (and could easily be triggered by trying to call load on a garbage non-string object)

```suggestion
            return compound
    elif isinstance(filename_or_object, str):
        continue
    else:
        raise ValueError('oops input isn't a thing we can recognize')
```

This assumes str or str-like is the only thing we can pass when referencing a file to load, I think this should work @justinGilmer?

The pathlib case shouldn't be an issue
```python3
>>> import pathlib
>>> x = pathlib.Path('/Users/mwt/foo.bar')
>>> isinstance(x.as_posix(), str)
True
```

Reply:
Yeah that seems like it would work @mattwthompson